docs/core/data-model/DataModel.md

```diff
@@ -0,0 +1,144 @@
+
+> [!IMPORTANT]
+> All automatically created resources are set to be `UPDATABLE`, meaning that they can be updated by pushing data updates, but not using a `SET` operation.
+
+#### Modifying the model
+
+When using a code first approach the model can always be updated and dynamically expanded. This means that the sending a new custom dto  extend the model to include any new services or resources as needed.
+
+A significant advantage of the code-first model is that it is not as easy to make mistakes when pushing updates, as the schema for the resources is defined in a class. This reduces the likelihood that the model for the providers will be incorrectly extended.
+
+### The explicit approach
+
+In some cases auto-generating the model gives insufficient control over the model that gets created. In this case a model can be explicitly generated in code by using the `SensinactModelManager` which gets passed to a sensiNact command when it is run. The following example code creates the same model as the code-first example above:
+
+```
+public class WeatherStationModel extends AbstractSensinactCommand<Void> {
+    @Override
+    protected Promise<Void> call(SensinactDigitalTwin twin, SensinactModelManager modelMgr,
+            PromiseFactory promiseFactory) {
+        modelMgr.createModel("WeatherStation")
+            .withService("power")
+                .withResource("status")
+                    .withType(Integer.class)
+                    .withValueType(ValueType.UPDATABLE)
+                    .build()
+                .build()
+            .withService("weather")
+                .withResource("temperature")
+                    .withType(Double.class)
+                    .withValueType(ValueType.UPDATABLE)
+                    .build()
+                .withResource("humidity")
+                    .withType(Double.class)
+                    .withValueType(ValueType.UPDATABLE)
+                    .buildAll();
+        return promiseFactory.resolved(null);
+    }
+}
+```
+
+
+### Importing models
+
+If you have created a suitable model for use in sensiNact then you can use the `SensinactModelManager` to import it directly.
+
+
+```
+public class WeatherStationModel extends AbstractSensinactCommand<Void> {
+    @Override
+    protected Promise<Void> call(SensinactDigitalTwin twin, SensinactModelManager modelMgr,
+            PromiseFactory promiseFactory) {
+        modelMgr.registerModel(new FileInputStream("myModel.xmi"));
+        return promiseFactory.resolved(null);
+    }
+}
+```
+
+## Whiteboard Resources
+
+In addition to pushed updates sensiNact allows resources to be registered using the whiteboard pattern. This allows for *pull based* data resources, where the gateway will query the connector to get or set an up to date value for a resource, and for action resources where the supplied function will be made available as an action to be called.
```

Comment:
Can we add an example or reference to the whiteboard pattern? It is not clear how this registration can be done.